tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -2466,6 +2466,44 @@ TYPED_TEST(LinalgBackendEigenAllTypesTest, SGMatrix_rank_update)
 	EXPECT_NEAR((A_eig - A_eig).norm(), 0, get_epsilon<TypeParam>());
 }
 
+TYPED_TEST(LinalgBackendEigenNonIntegerTypesTest, SGVector_sqrt)
+{
+	SGVector<TypeParam> data{1.0, 2.0, 3.0};
+	auto result = linalg::sqrt(data);
+
+	for (index_t i = 0; i < data.vlen; ++i)
+	EXPECT_NEAR(result[i], std::sqrt(data[i]), get_epsilon<TypeParam>());
+}
+
+TYPED_TEST(LinalgBackendEigenNonIntegerTypesTest, SGVector_sqrt_in_place)
+{
+	SGVector<TypeParam> data{1.0, 2.0, 3.0};
+	SGVector<TypeParam> result(data.vlen);
+	linalg::sqrt(data);
+
+	for (index_t i = 0; i < result.vlen; ++i)
+		EXPECT_NEAR(result[i], std::sqrt(data[i]), get_epsilon<TypeParam>());
```

Comment:
you probably need to copy the container no?

Reply:
oh, sorry.